LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -30,96 +36,103 @@ internal sealed class ConcentratorDeduplication<T> :
 
         public ConcentratorDeduplication(
             IMemoryCache cache,
-            WebSocketWriterRegistry<StationEui, string> socketRegistry,
-            ILogger<IConcentratorDeduplication<T>> logger)
+            IDeduplicationStrategyFactory deduplicationStrategy,
+            ILogger<IConcentratorDeduplication> logger)
         {
             this.cache = cache;
-            this.socketRegistry = socketRegistry;
+            this.deduplicationStrategy = deduplicationStrategy;
             this.logger = logger;
         }
 
-        public bool ShouldDrop(T frame, StationEui stationEui)
+        public Result CheckDuplicate(LoRaRequest loRaRequest, LoRaDevice? loRaDevice)
         {
-            if (frame == null) throw new ArgumentNullException(nameof(frame));
+            _ = loRaRequest ?? throw new ArgumentNullException(nameof(loRaRequest));
 
-            var key = CreateCacheKey(frame);
+            var key = CreateCacheKey(loRaRequest);
+            var stationEui = loRaRequest.StationEui;
 
             StationEui previousStation;
             lock (this.cache)
             {
                 if (!this.cache.TryGetValue(key, out previousStation))
                 {
-                    AddToCache(key, stationEui);
-                    return false;
+                    _ = this.cache.Set(key, stationEui, new MemoryCacheEntryOptions()
+                    {
+                        SlidingExpiration = DefaultExpiration
+                    });
+                    return Result.NotDuplicate;
                 }
             }
 
-            if (previousStation == stationEui)
+            if (LoRaRequest.RequiresConfirmation(loRaRequest) && previousStation == stationEui)
             {
-                // considered as a resubmit
-                this.logger.LogDebug("Message received from the same EUI {StationEui} as before, will not drop.", stationEui);
-                return false;
+                this.logger.LogDebug($"Message received from the same EUI {stationEui} as before, will not drop.");
```

Comment:
message here is not clearly indicating that this is a confirmed resubmit and the same station is handling it.

Reply:
Not sure what you would like to see here. I added the "confirmedResubmit" c29c90d54bddea7d2ff1caf0f69f027f0a14df33 as part of the log because I understand this is the keyword you were missing although imo "confirmed resubmit" implies that message was confirmed already (which is not the case).